master/internal/stream/system.go

```diff
@@ -167,6 +239,26 @@ func (ps PublisherSet) Websocket(socket *websocket.Conn, c echo.Context) error {
 		streamer.Close()
 	}()
 
+	// detect & handle permission change
+	go func() {
+		if err := AuthZProvider.Get().WaitForPermissionChange(); err != nil {
+			// failed to setup permission change detection, restarting publisher set
+			var errorPrefix string
+			switch {
+			case authz.IsPermissionChangeError(err):
+				errorPrefix = "permission change detected"
+			default:
+				errorPrefix = "error occurred while waiting for permission changes"
+			}
+			log.Errorf("%s, restarting publisher set: %s", errorPrefix, err)
+			for _, err := range ps.Restart() {
```

Comment:
The placement of this gofunc inside WebSocket makes me think that the publisher set will get .Restart()ed N times when a change happens while N clients are connected, where N might be in the thousands.  That seems odd.

Reply:
I could move this logic into `NewPublisherSet`, so when a new PublisherSet is created it also begins listening for permission changes. 

That way not every single client is watching; but rather the PublisherSet itself is. I also like this because the current implementation (with `WaitForPermissionChange`) runs into the same issue as before w/ opening up a lot of unnecessary pq.Listeners

Reply:
Moved this into ps.Start()